schemas/iso19115-3.2018/src/main/resources/config-spring-geonetwork.xml

```diff
@@ -45,6 +45,10 @@
                       cit:name/gco:CharacterString = '{{name}}']
                     /cit:applicationProfile/gco:CharacterString/text())"/>
         </bean>
+        <bean class="org.fao.geonet.kernel.schema.SavedQuery">
+			    <property name="id" value="doi-get"/>
+			    <property name="xpath" value="//cit:CI_OnlineResource[cit:protocol/gco:CharacterString = 'DOI']/cit:linkage/cit:URL/text()"/>
```

Comment:
`linkage` is a `CharacterString` in ISO19115-3. See
https://github.com/geonetwork/core-geonetwork/blob/main/schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/schema/standards.iso.org/19115/-3/cit/2.0/citation.xsd#L288

```xml
<property name="xpath" 
          value="//cit:CI_OnlineResource[cit:protocol/gco:CharacterString = 'DOI']/cit:linkage/cit:CharacterString/text()"/>
```

Reply:
Thanks for fixing this. Even if it was working, you are perfectly right this is the correct way.